src/transformers/models/llava_next/image_processing_llava_next.py

```diff
@@ -603,6 +621,7 @@ def preprocess(
             pixel_values = np.array(pixel_values)
             new_images.append(pixel_values)
 
-        data = {"pixel_values": new_images, "image_sizes": image_sizes}
-
-        return BatchFeature(data=data, tensor_type=return_tensors)
+        padded_images = self._pad_for_batching(new_images)
```

Comment:
As mentioned before - this should be controlled by a `do_pad` flag

Reply:
Ah, I just realized there is a special flag in some image processors